kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -311,29 +329,68 @@ public void initiateConnect(HostPort remote, List<FilterAndInvoker> filters) {
             pipeline.addFirst("networkLogger", new LoggingHandler("io.kroxylicious.proxy.internal.UpstreamNetworkLogger"));
         }
 
-        virtualCluster.getUpstreamSslContext().ifPresent(c -> pipeline.addFirst("ssl", c.newHandler(outboundChannel.alloc(), remote.host(), remote.port())));
-
-        connectFuture.addListener(future -> {
+        Future<?> kafkaAvailableFuture = virtualCluster.getUpstreamSslContext().<Future<?>> map(c -> {
+            final SslHandler handler = c.newHandler(outboundChannel.alloc(), remote.host(), remote.port());
+            pipeline.addFirst("ssl", handler);
+            exceptionHandler.registerExceptionResponse(SSLHandshakeException.class, throwable -> {
```

Comment:
`SSLHandlshakeException` has subclasses, which means the `exceptionHandler` won't invoke the lambda if `handleException()` is called with an instance of one of those subclasses. I suspect that's not what we want?